policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/request/PolicyResponse.java

```diff
@@ -0,0 +1,293 @@
+        }
+
+        /**
+         * Adds the user context information to the message.
+         */
+        interface IContext {
+            /**
+             * Adds the user context information.
+             *
+             * @param context user context for the request.
+             * @return interface {@link IUser} for the next step in the build.
+             */
+            default IUser withContext(Context context) {
+                return withContextNode(MAPPER.valueToTree(context));
+            }
+
+            /**
+             * Adds the user context information.  Uses a JsonNode string form of the information.
+             *
+             * @param context user context for the request.
+             * @return interface {@link IUser} for the next step in the build.
+             */
+            IUser withContextNode(JsonNode context);
+        }
+
+        /**
+         * Adds the user information to the message.
+         */
+        interface IUser {
+            /**
+             * Adds the user user information.
+             *
+             * @param user for the request.
+             * @return class {@link IResource} for the next step in the build.
+             */
+            default IResource withUser(User user) {
+                return withUserNode(MAPPER.valueToTree(user));
+            }
+
+            /**
+             * Adds the user user information.  Uses a JsonNode string form of the information.
+             *
+             * @param user for the request.
+             * @return class {@link IResource} for the next step in the build.
+             */
+            IResource withUserNode(JsonNode user);
+        }
+
+        /**
+         * Adds the resource to this message.
+         */
+        interface IResource {
+            /**
+             * Adds the resource that has been requested to access.
+             *
+             * @param resource that is requested to access
+             * @return class {@link IRules} for the next step in the build.
+             */
+            default IRules withResource(Resource resource) {
+                return withResourceNode(MAPPER.valueToTree(resource));
+            }
+
+            /**
+             * Adds the resource that has been requested to access.  Uses a JsonNode string form of the information.
+             *
+             * @param resource that is requested to access
+             * @return class {@link IRules} for the next step in the build.
+             */
+            IRules withResourceNode(JsonNode resource);
+        }
+
+        /**
+         * Adds the rules to this message.
+         */
+        interface IRules {
+            /**
+             * Adds the rules that has apply to this request.
+             *
+             * @param rules that apply to this request.
+             * @return class {@link PolicyResponse} for the completed class from the builder.
+             */
+            PolicyResponse withRule(Rules rules);
```

Comment:
withRule(Rules rules)? withRules(Rules rules) surely?

Reply:
In fact I'd add another method along the lines of PolicyResponse withRule(Stream<Rule> rules) in which you construct a Rules object.

Reply:
am not sure on the need for a withRules methods that takes in a stream of rules, I don't think it is needed at this point. That might change when we take a closer look at how we merge rules set at different levels of the hierarchy, but that is a problem for another day. I do agree with the withRule method needing pluralising though

Reply:
Fine, just pluralise.